### ISSUE TITLE
Trim deep indentation

### ISSUE THREAD

**Issue by will-lynas** [yetone/avante.nvim#465]
When working with deeply nested code, the sidebar view is not easy to read. I think it would be better if the code was left-trimmed so that there are no columns of pure whitespace.

<img width="455" alt="image" src="https://github.com/user-attachments/assets/cc2b71fd-79d3-43ca-b598-718868766282">

**Comment by aarnphm** [comment]
this is actually intended, otherwise the LLM won't generate correct code.

We need to preserve indentation somehow. Maybe for rendering on sidebar we can skip indentation on view 🤔 

Feel free to propose something here.

**Comment by will-lynas** [comment]
Yeah that's what I mean. Keep the original response but render a trimmed one.

**Comment by aarnphm** [comment]
feel free to contribute. Check sidebar.lua for this.

**Comment by will-lynas** [comment]
This is harder than I initially thought because of the token streaming.

And the fact that the code snippets are extracted directly from the content that is rendered in the sidebar: https://github.com/yetone/avante.nvim/blob/713cb403d0e3c3b9b8b9cacb1ca0ff5d6fc35a7d/lua/avante/sidebar.lua#L316

I guess a really hacky way to do this is:
- At each sidebar update, try to find complete code blocks, then strip their whitespace.
- Record the number of characters stripped (by writing the number next to the code block in the actual text? - really hacky)
- Reconstruct the whitespace when parsing out the code blocks for applying them.

But as a more general solution, what about this:
- Parse the LLM response into some intermediate data structure that has more structure for code blocks.
- Render the sidebar using this internal representation.
- Generate the code diffs using the internal representation.

This way, we are free to change the formatting of the code in the sidebar without it affecting the actual diff.

I'm not sure how to handle partially complete code blocks. I can think of either:
- Render the direct LLM response, then parse it into a proper formatted code block once it is complete. But this will mean the sidebar will be changing content that has already been displayed which could be a bit distracting.
- Buffer code blocks until they are complete and can be rendered properly. Could be annoying for large code blocks.

What are your thoughts?
